src/indexes.ts

```diff
@@ -327,10 +327,10 @@ class Index<T extends Record<string, any> = Record<string, any>> {
    * @param parameters - Parameters applied on a document
    * @returns Promise containing Document response
    */
-  async getDocument<T = Record<string, any>>(
+  async getDocument<D extends Record<string, any> = T>(
```

Comment:
Why do we need to keep extending here? Maybe there is a case that I don't see. Is `D = T` not enough ?

Reply:
I see, but imagine I want to get `Movie` documents

```js
type Movie = {
   id: number
   name: string
}
```

Then, overwrite the type of `index` by doing like you said `client.index('x').getDocument<Movie>(1)`.

In this case `Movie` is doing to be `D`. But what does `D` needs to be extended from `Record<string, any>` ? I'm not expecting any other field in my `Movie`. 

Sorry if I'm misunderstanding something

Reply:
your type `Movie` already extends `Record<string, any>`

it's valid to call `getDocument<Movie>()` with your type example

Reply:
Ah I see! Thanks :)

Reply:
In the same vain, should we have added that as well in `search` ?

https://github.com/meilisearch/meilisearch-js/blob/b1ed904adabbe4f65ee749a3df92ef2f773c4211/src/indexes.ts#L102

```js
 async searchGet<D extends Record<string, any>> = T>( 
```

Reply:
yes

Reply:
https://github.com/meilisearch/meilisearch-js/issues/1425